src/language/class-modifiers.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Class modifiers
+description: Modifier keywords for class declarations to control external library access.
+---
+
+Class modifiers control how a class or a mixin can be implemented, extended,
+or mixed in. 
+
+The full set of modifiers that can appear before a class declaration are `base`,
+`final`, `interface`, `sealed`, `abstract`, and `mixin`. 
+Only the `base` modifier can appear before a mixin declaration. The modifiers do
+not apply to other declarations like `enum`, `typedef`, or `extension`.
+
+When deciding whether or not to use class modifiers, consider the following questions:
+1. Is [any access](#no-modifier) fine?
+2. If not, what do you want to restrict?
+    - [Implementation](#base)?
+    - [Inheritance](#interface)?
+    - [Both](#final)?
+    - [Subtyping](#sealed) by outside libraries at all?
+3. Then, does it need to be [abstract](#abstract)?
+4. And lastly, should it be a [class, mixin, or mixin class](#class-mixin-or-mixin-class)?
+
+## No modifier
+
+Use a `class` or `mixin` declaration without a modifier to **allow unrestricted
+access** to outside libraries. The standard behaviors apply:
+
+- [Construct][] new instances of a class
+- [Extend][] a class to create a new subtype
+- [Implement][] a class's interface
+- [Mix in][] a mixin or mixin class
+
+## `base`
+
+Use the `base` modifier to **disallow implementation** of a class or mixin.
+A base class can be extended and constructed. This applies trasitively to all
```

Comment:
This also sounds like a `base` class cannot be `abstract`, but that's an orthogonal modifier. An `abstract base class` cannot be constructed.

I'd not discuss construction in the section about `base`, because it's mostly unrelated.

Reply:
Do you think _all_ the statements about what _isn't_ restricted for each modifier should just be removed? Like this line, or "An interface class _can_ be implemented and constructed.", etc.